app/ts/LiquidityProvidingUI/Liquidity.tsx

```diff
@@ -15,13 +15,61 @@ import { DaiNameAndSymbol } from '../SharedUI/tokens.js'
 import { useEffect } from 'preact/hooks'
 import { BigInputBox } from '../SharedUI/BigInputBox.js'
 import { getAugurConstantProductMarketRouterAddress, isAugurConstantProductMarketRouterDeployed } from '../utils/augurDeployment.js'
+import { tickToZeroToOnePrice, zeroOnePriceToTick } from '../utils/uniswapUtils.js'
+
+interface LiquidityTokensProps {
+	liquidityTokens: OptionalSignal<Awaited<ReturnType<typeof getUserLpTokenIdsAndBalancesForMarket>>>
+	maybeWriteClient: OptionalSignal<WriteClient>
+	marketData: OptionalSignal<MarketData>
+	currentTimeInBigIntSeconds: Signal<bigint>
+}
+
+const LiquidityTokens = ({ liquidityTokens, maybeWriteClient, marketData, currentTimeInBigIntSeconds }: LiquidityTokensProps) => {
+	if (liquidityTokens.deepValue === undefined) {
+		return <section>
+			<h3>Liquidity Positions</h3>
+			<p> loading...</p>
+		</section>
+	}
+	if (liquidityTokens.deepValue.length === 0) {
+		return <section>
+			<h3>Liquidity Positions</h3>
+			<p> No Positions</p>
+		</section>
+	}
+	const closePosition = async (tokenId: bigint) => {
+		if (maybeWriteClient.deepValue === undefined) return
+		if (marketData.deepValue === undefined) return
+		if (liquidityTokens.deepValue === undefined) return
+		const amountNoMin = 0n // TODO, add slippage calculations
+		const amountYesMin = 0n // TODO, add slippage calculations
```

Comment:
Should we add an issue to take care of these?

Reply:
added